### PR TITLE
Fix base styles

### PR DESCRIPTION
**Pull request by cole-sanderson**
Move styles that was in `_levels.scss` into proper place in base styles. 

 **JIRA**: https://mobify.atlassian.net/browse/AMP-51
 **Linked PRs**: n/a

## Changes
- remove `_level.scss`
- move styles that was in `_level.scss` into proper place in base styles. 
- remove `!important` for `[hidden]`
- remove `!important` in button on PDP, and use add classes instead.
- wrap styles in template class,`t-onboarding`.

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Check all pages in MP to make sure that it looks same.

### CODE REVIEW

**Comment by cole-sanderson** [comment]
@yourpalsonja All resets styles now live in base folder in few different files.

File: web/app/styles/base/_forms.scss [hunk 1]
```diff
@@ -6,6 +6,16 @@ $base__radio-checkmark: 'data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2
 
 // General Form Elements
 // ---
+//
+// 1. Address Firefox 4+ setting `line-height` on `input` using `!important` in
+// the UA stylesheet.
+// 2. Remove padding so people aren’t caught out if they zero out fieldsets.
+// 3. Remove inner padding and border in Firefox 4+.
+
```
Comment (jeffkamo) [inline web/app/styles/base/_forms.scss]: 🔫  (nitpick: can delete this empty line)

File: web/app/styles/base/_forms.scss [hunk 2]
```diff
@@ -90,13 +133,25 @@ input[type="search"] {
 //    styling, remove both these lines.
 // 2. Simulate the position of the down-arrow as if it were a Icon component in
 //    a button.
+// 3. Address inconsistent `text-transform` inheritance for `button` and `select`.
+// All other form control elements do not inherit `text-transform` values.
+// Correct `button` style inheritance in Firefox and Opera. Correct `select`
+// style inheritance in Firefox.
```
Comment (jeffkamo) [inline web/app/styles/base/_forms.scss]: The above comments should be indented to match the position of # 3

File: web/app/styles/base/_forms.scss [hunk 3]
```diff
@@ -90,13 +133,25 @@ input[type="search"] {
...
+
```
Comment (jeffkamo) [inline web/app/styles/base/_forms.scss]: 🔫 (nitpick: can delete this empty line)

File: web/app/styles/base/_forms.scss [hunk 4]
```diff
@@ -69,19 +99,32 @@ textarea,
...
 // Search input
 // ---
 //
 // These properties must be set with a slightly higher specificity for search
 // inputs because Normalize's defaults are a bit too specific
+//
+// 1. Address `appearance` set to `searchfield` in Safari and Chrome.
+// 2. Remove inner padding and search cancel button in Safari and Chrome on OS X.
+// Safari (but not Chrome) clips the cancel button when the search input has
+// padding (and `textfield` appearance).
```
Comment (jeffkamo) [inline web/app/styles/base/_forms.scss]: The above comments should be indented to match the position of # 2

File: web/app/styles/base/_forms.scss [hunk 5]
```diff
@@ -190,25 +252,52 @@ select {
...
 // ---
+//
+// 1. Address margins set differently in Firefox 4+, Safari, and Chrome.
+// 2. Correct color not being inherited. Known issue: affects color of disabled
+//    elements.
+// 3. Correct font properties not being inherited.
+// 4. Address inconsistent `text-transform` inheritance for `button` and `select`.
+// All other form control elements do not inherit `text-transform` values.
+// Correct `button` style inheritance in Firefox and Opera. Correct `select`
+// style inheritance in Firefox.
```
Comment (jeffkamo) [inline web/app/styles/base/_forms.scss]: The above comments should be indented to match the position of # 4

File: web/app/styles/base/_general.scss [hunk 6]
```diff
@@ -6,27 +6,123 @@
...
+
+
+// Grouping content
+// ---
+//
+// 1. Address margin not present in Safari.
+// 2. Contain overflow in all browsers.
+// 3. Address odd `em`-unit font size rendering in all browsers.The duplication
+// of `monospace` is intentional
+// ([Source](http://en.wikipedia.org/wiki/User:Davidgothberg/Test59)).
```
Comment (jeffkamo) [inline web/app/styles/base/_general.scss]: This indentation should match that of # 3

File: web/app/styles/base/_general.scss [hunk 7]
```diff
@@ -6,27 +6,123 @@
...
+    font-size: 1em; // 3
 }
 
+
+// New HTML5 elements
+// ---
+
+// Correct `block` display not defined for `main` in mobile Safari 6 and
+// Android 4.3.
+// Correct `block` display not defined for `menu`.
```
Comment (jeffkamo) [inline web/app/styles/base/_general.scss]: Is this a new paragraph? It should have an empty line separating it from the prior paragraph.

File: web/app/styles/base/_general.scss [hunk 8]
```diff
@@ -6,27 +6,123 @@
...
+
+
 // Hidden
 // ---
 //
 // Hide visually and from screen readers. The html5 `hidden` attribute is
-// preferred to a `u-hidden` class because it exactly matches our intent and
-// has machine-readable semantics.
+// preferred to a `u-visually-hidden` class because it exactly matches our
+// intent and has machine-readable semantics.
```
Comment (jeffkamo) [inline web/app/styles/base/_general.scss]: Hmm, the comment above is actually slightly troubling now. With the removal of `!important`, the below style can't be guaranteed to work in all conditions... but `u-visually-hidden` can be guaranteed to work. Maybe we should just remove the "preferred" part of the above comment.

**Pull request merged**